apps/indexer/lib/indexer/fetcher/celo_voter_votes.ex

```diff
@@ -0,0 +1,133 @@
+defmodule Indexer.Fetcher.CeloVoterVotes do
+  @moduledoc """
+  Fetches Celo voter votes.
+  """
+  use Indexer.Fetcher
+  use Spandex.Decorators
+
+  require Logger
+
+  alias Explorer.Celo.{AccountReader, ContractEvents}
+  alias Explorer.Chain
+  alias Explorer.Chain.{Block, Hash}
+  alias Explorer.Chain.CeloVoterVotes, as: CeloVoterVotesChain
+
+  alias ContractEvents.Election.ValidatorGroupVoteActivatedEvent
+
+  alias Indexer.BufferedTask
+  alias Indexer.Fetcher.Util
+
+  @behaviour BufferedTask
+
+  @spec async_fetch([%{block_hash: Hash.Full.t(), block_number: Block.block_number()}]) :: :ok
+  def async_fetch(blocks) when is_list(blocks) do
+    filtered_blocks =
+      blocks
+      |> Enum.filter(&(rem(&1.block_number, 17280) == 0))
+
+    BufferedTask.buffer(__MODULE__, filtered_blocks)
+  end
+
+  @doc false
+  def child_spec([init_options, gen_server_options]) do
+    init_options_with_polling =
+      init_options
+      |> Keyword.put(:poll, true)
+      |> Keyword.put(:poll_interval, :timer.minutes(60))
+      |> Keyword.put(:max_concurrency, 1)
+      |> Keyword.put(:max_batch_size, 1)
+
+    Util.default_child_spec(init_options_with_polling, gen_server_options, __MODULE__)
+  end
+
+  @impl BufferedTask
+  def init(initial, reducer, _json_rpc_named_arguments) do
+    {:ok, final} =
+      Chain.stream_blocks_with_unfetched_voter_votes(initial, fn block, acc ->
```

Comment:
I have a feeling that the whole thing could produce wrong results if fetcher fails for some blocks in the beginning, the failing blocks will be requeued and processed after the rest of the queue that would break the logic as it relies on strict order of block processing.